src/modelmanager.cpp

```diff
@@ -877,6 +877,7 @@ void cleanerRoutine(uint32_t resourcesCleanupInterval, FunctorResourcesCleaner&
 }
 
 void ModelManager::cleanupResources() {
+    std::unique_lock resourcesLock(resourcesMtx);
```

Comment:
Here copy the elements to be removed to temporary container. Then erase those elements from original container.
Afterwards unlock the mutex, then remove temporary container - it should fall out of the scope & remove itself

Reply:
This is to avoid potential long Custom Node Library INtermal Manager deinitialize, which could block the config updating thread.